### PR TITLE
Fullstack model implementation

### PR DESCRIPTION
**Pull request by Kris2ffer**

### CODE REVIEW

File: RestApi/Sepes.RestApi/Services/AzureService.cs [hunk 1]
```diff
@@ -146,6 +150,49 @@ await _azure.NetworkSecurityGroups
...
+            .DefineRule("AllowVnetoutBound2")
+            .AllowOutbound()
+            .FromAddress("VirtualNetwork")
+            .FromAnyPort()
+            .ToAddress("VirtualNetwork")
+            .ToAnyPort()
+            .WithAnyProtocol()
+            .WithPriority(4000)
+            .Attach()
+            .ApplyAsync();
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Services/AzureService.cs]: This have be that builder pattern taken to its logical extreme. Should be split up into something smaller.

File: RestApi/Sepes.RestApi/Services/PodService.cs [hunk 2]
```diff
@@ -115,46 +115,37 @@ private async Task ManageNetworkSecurityGroup(Pod newPod, Pod based)
...
             }
 
+
             // Apply network security group to subnet
             Console.WriteLine($"#### ApplySecurityGroup() {newPod.resourceGroupName}, {nsgName}, {newPod.subnetName}, {newPod.networkName}");
             await _azure.ApplySecurityGroup(newPod.resourceGroupName, nsgName, newPod.subnetName, newPod.networkName);
 
             // Delete old nsg
             if (nsgName == nsgNameDefault && nsgNames.Contains(nsgNameDef2)) {
+                Console.WriteLine("#### Deleting old nsg "+nsgNameDef2);
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Services/PodService.cs]: Console.WriteLine should not be in master. If debug remove. If not log it using logger.

File: RestApi/Sepes.RestApi/Services/PodService.cs [hunk 3]
```diff
@@ -115,46 +115,37 @@ private async Task ManageNetworkSecurityGroup(Pod newPod, Pod based)
...
                 await _azure.DeleteSecurityGroup(nsgNameDef2, newPod.resourceGroupName);
             }
             else if (nsgName == nsgNameDef2 && nsgNames.Contains(nsgNameDefault)) {
+                Console.WriteLine("#### Deleting old nsg "+nsgNameDefault);
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Services/PodService.cs]: Console.WriteLine should not be in master. If debug remove. If not log it useing logger.

File: RestApi/Sepes.RestApi/Services/StudyService.cs [hunk 4]
```diff
@@ -74,16 +76,19 @@ private Study UpdatePods(Study based, Study study)
                         // Update list of pod
                         // generate new pod with id
                         Pod newPod = pod.NewPodId(numberOfPods++);
+                        newPod = newPod.AddUserList(study.suppliers);
                         var newPods = study.pods.Remove(pod).Add(newPod);
                         study = study.ReplacePods(newPods);
 
+                        Console.WriteLine("Create pod with pod id "+newPod.id);
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Services/StudyService.cs]: Another Console.WriteLine

**Review by gjersvik — CHANGES_REQUESTED** [review]
Please remove the Console.WriteLine from the code. Also not required but try to spit the mega builder.

**Review by gjersvik — APPROVED** [review]

**Review by Unicus-Tom — APPROVED** [review]

**Review by Unicus-Tom — APPROVED** [review]

**Pull request merged**